### ISSUE TITLE
sub-sampling using correlation distance

### ISSUE THREAD

**Issue by AndreaGarciaJuan** [euroargodev/DMQC-PCM#18]
- [x] distance calculation (**[sklearn.metrics.pairwise.haversine_distances](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.haversine_distances.html)**) for all points first
- [x] iterations until all points are far enough
- [x] use mask
- [x] how long it takes?

**Comment by AndreaGarciaJuan** [comment]
In the new version of the 'get_regulargrid_dataset' function I calculate the distance matrix only for the profiles around the random profile I choose in each iteration. There is no problems with RAM memory because the matrices are little and it do not take a lot of time to compute:

float in Gulf Stream (8793 profiles): **13s** 
float in Southern Ocean (16435 profiles): **30s**
float in Agulhas Current (18391 profiles): **30s**

Example of results: 
![Capture d'écran de 2022-01-12 11-32-16](https://user-images.githubusercontent.com/59824937/149124361-7c2065b3-a444-4965-ae58-3932b983c41d.png)

**Comment by AndreaGarciaJuan** [comment]
Calculating the BIC with this subsampling method gives "reasonable" computation time (with 10 runs and testing k=[1:15]:
float in Gulf Stream (8793 profiles): **4 min**
float in Southern Ocean (16435 profiles): **10 min**
float in Agulhas Current (18391 profiles): **9 min**

exemple (Agulhas Current): 
![Capture d'écran de 2022-01-13 11-27-19](https://user-images.githubusercontent.com/59824937/149313275-3d7abe17-976b-410f-954f-6450ebc6ad51.png)
